### PR TITLE
Implement Supervisor that works with the Consensus Layer 

### PR DESCRIPTION
**Pull request by agunde406**
This PR adds a Supervisor that will handle pending notifications
that have been received from the consensus layer. The supervisor
currently handles each notification in the same way that the
CommandNotifyObserver did.

When a supervisor notification is added to state the Supervisor
should be notified. This must happen after the commands have been
executed, as such the notifier will be added to the timer handler.

### CODE REVIEW

File: services/scabbard/libscabbard/src/store/scabbard_store/diesel/operations/list_ready_services.rs [hunk 1]
```diff
@@ -108,6 +111,22 @@ where
                     })?,
             );
 
+            ready_services.append(
```
Comment (isabeltomb) [inline services/scabbard/libscabbard/src/store/scabbard_store/diesel/operations/list_ready_services.rs:114]: should this query also check to make sure that the services with unexecuted supervisor notifications are in the finalized state?

File: services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        &self.action_id
+    }
+
+    /// Returns the notification_type for the notficiation
+    pub fn notification_type(&self) -> &SupervisorNotificationType {
+        &self.notification_type
+    }
+
+    /// Returns the value for the notficiation
+    pub fn request_for_vote_value(&self) -> &Option<Vec<u8>> {
```
Comment (vaporos) [inline services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs:48]: Consider include this in the enum to avoid the Option

Comment (agunde406) [reply services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs]: Updated to remove request_for_vote_value and use `SupervisorNotificationType::ParticipantRequestForVote { value: Vec<u8> },` instead

File: services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+        &self.request_for_vote_value
+    }
+
+    /// Returns the executed_at time for the notficiation
```
[the remark this answers is not in the document]
Comment (agunde406) [reply services/scabbard/libscabbard/src/store/scabbard_store/supervisor.rs]: fixed

File: services/scabbard/libscabbard/src/store/scabbard_store/diesel/operations/list_ready_services.rs (hunk 1 above)
Comment (agunde406) [reply services/scabbard/libscabbard/src/store/scabbard_store/diesel/operations/list_ready_services.rs:114]: added

**Review by isabeltomb — DISMISSED** [review]

**Review by vaporos — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**